Calendar Report

lelemm opened the pull request:
A new report type where data shows in a calendar format`:


https://github.com/user-attachments/assets/229e42e9-3b74-435f-9b00-92c14a273e8a

lelemm commented on packages/desktop-client/src/components/reports/reportRanges.ts:
This is a fix because when offset is one, its returning 2 months instead of one.

lelemm commented on packages/desktop-client/src/components/transactions/TransactionsTable.jsx:
```diff
@@ -1504,7 +1520,7 @@ const Transaction = memo(function Transaction({
                 focused={true}
                 clearOnBlur={false}
-                showSplitOption={!isChild && !isParent}
+                showSplitOption={!isChild && !isParent && allowSplitTransaction}
```
out of the scope of the PR

lelemm commented on packages/desktop-client/src/components/reports/graphs/CalendarGraph.tsx:
```diff
@@ -0,0 +1,290 @@
+        }}
+      >
+        {data.map((day, index) =>
+          !isSameMonth(day.date, startOfMonth(start)) ? (
+            <View key={index} />
+          ) : (
+            <Tooltip
+              key={index}
+              content={
```
done

lelemm commented on packages/desktop-client/src/components/reports/spreadsheets/calendar-spreadsheet.ts:
```diff
@@ -0,0 +1,211 @@
+      }[];
+    }) => void,
+  ) => {
+    const { filters } = await send('make-filters-from-conditions', {
+      conditions: conditions.filter(cond => !cond.customName),
+    });
```
this looks incomplete, the variable "filters" is used on line 64, and its not inside the try

lelemm commented on packages/desktop-client/src/style/themes/development.ts:
```diff
@@ -213,3 +213,5 @@ export const floatingActionBarText = colorPalette.navy50;
 export const tooltipBackground = colorPalette.navy50;
 export const tooltipBorder = colorPalette.navy150;
+
+export const calendarCellBackground = colorPalette.navy900;
```
they are used for different calendars

lelemm commented on packages/desktop-client/src/components/reports/reports/CalendarCard.tsx:
```diff
@@ -0,0 +1,542 @@
+    return measurements;
+  }, []);
+
+  const debouncedResizeCallback = useRef(
+    debounce(() => {
+      const measurements = measureMonthFormats();
+      const containerWidth = monthNameContainerRef.current?.clientWidth ?? 0;
+
+      const suitableFormat = measurements.find(m => containerWidth > m.width);
+      if (suitableFormat) {
+        if (
+          monthNameContainerRef.current &&
+          containerWidth > suitableFormat.width
+        ) {
+          setMonthNameFormats(prev => {
+            const newArray = [...prev];
+            newArray[index] = suitableFormat.format;
+            return newArray;
+          });
+
+          setMonthNameVisible(true);
+          return;
+        }
+      }
+
+      if (
+        monthNameContainerRef.current &&
+        monthNameContainerRef.current.scrollWidth >
+          monthNameContainerRef.current.clientWidth
+      ) {
+        setMonthNameVisible(false);
+      } else {
+        setMonthNameVisible(true);
+      }
+    }, 20),
+  );
```
debouncedResizeCallback is used with useResizeObserver:

import { useRef, useCallback } from 'react';

export function useResizeObserver(
  func: (contentRect: DOMRectReadOnly) => void,
): (el: unknown) => void {
  const observer = useRef(null);
  if (!observer.current) {
    observer.current = new ResizeObserver(entries => {
      func(entries[0].contentRect);
    });
  }

  const elementRef = useCallback(el => {
    observer.current.disconnect();
    if (el) {
      observer.current.observe(el, { box: 'border-box' });
    }
  }, []);

  return elementRef;
}

can you expand your logic so I can understand what you mean

joel-jeremy commented:
Some observations:
- Does this need its own experimental feature flag?
  - Currently if the customizable dashboards flag is off, there is no way to create a calendar report
- There is a bug where if you keep on swiping up the collapsible transaction list, it also keeps on moving up

lelemm commented:
> Could the dashboard tooltips say the month instead of just "Day"? So "Nov 13" instead of "Day 13" ![image](https://private-user-images.githubusercontent.com/28542559/387264534-6482de7b-86e0-4819-880b-d6abdfcbdfa3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE5NDQ1NDMsIm5iZiI6MTczMTk0NDI0MywicGF0aCI6Ii8yODU0MjU1OS8zODcyNjQ1MzQtNjQ4MmRlN2ItODZlMC00ODE5LTg4MGItZDZhYmRmY2JkZmEzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTE4VDE1MzcyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY3ZGQzOTAxZmE2YjYxMmNiMDQ5YzhmNDY2Zjk5ZDFhMzVkMmY4OWQzYjc4ODcyNTM2NWJjYWJlNDNmZThlMmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.sk4VCQb-UKGYEDKW3hPhU1_EyS9ZNaHgWZnUQP7F10w)

I did that based on the monthly spending.
![image](https://github.com/user-attachments/assets/6626d829-06b2-4b9a-9b7a-d5c1263c2cc1)
I will change to Month + day

lelemm commented:
done @youngcw. 
Limited to 4, then it adds a scrollbar, also changed the tooltip title

youngcw commented:
The scroll bar seems a bit thick, but other than that looks great!

lelemm commented:
> The scroll bar seems a bit thick, but other than that looks great!

that's the default horizontal scrollbar style:
![image](https://github.com/user-attachments/assets/c6171390-a8b5-45e8-bbf2-b2250ff41d21)
![image](https://github.com/user-attachments/assets/9eec77ba-0289-4ce4-b9cc-76e2d573c820)

youngcw commented:
The popovers for the calendar dashboard card are different than the other reports.  But to be honest using that style looks nicer.  Maybe the others should be updated instead of this changed.

When selecting different days from the full report, the calendar at the top will move up and down since not all days have both income and spending.  It would be nice if that stayed consistent and had the same sizing no matter if the days had both in and out, or just one.

It would also be nice if you could select a range of days when looking at the full report.  Maybe the ability to ctrl-click and shift-click on days to make selections and ranges.  That may be for a different PR if that is too much for now.

If may be nice from a UX view to unselect the selected day in the full report if you click outside the calendar.

lelemm commented:
> The popovers for the calendar dashboard card are different than the other reports. But to be honest using that style looks nicer. Maybe the others should be updated instead of this changed.

Yeah, the other reports use recharts component, the tooltip is part of the component, the tooltip i'm using here is the standard actual tooltip component. That's why its different.

> 
> When selecting different days from the full report, the calendar at the top will move up and down since not all days have both income and spending. It would be nice if that stayed consistent and had the same sizing no matter if the days had both in and out, or just one.

I think I don't get this one. Can you record/screenshot what you mean?

> 
> It would also be nice if you could select a range of days when looking at the full report. Maybe the ability to ctrl-click and shift-click on days to make selections and ranges. That may be for a different PR if that is too much for now.
> 

That's a nice idea, but I guess It would take rework to make this feasible, may be another PR would be better. 

> It may be nice from a UX view to unselect the selected day in the full report if you click outside the calendar. Right now you have to remove the filter that gets generated.

yeah, nice one. I will do it

youngcw commented:
This is what I mean.  The calendar shifts around as you click different days

https://github.com/user-attachments/assets/e04a6971-af29-4c19-93c5-f4cba7a45518

lelemm commented:
> This is what I mean. The calendar shifts around as you click different days
> 
>  cal.mp4

got it

lelemm commented:
> > It may be nice from a UX view to unselect the selected day in the full report if you click outside the calendar. Right now you have to remove the filter that gets generated.
> 
> yeah, nice one. I will do it

turns out, not easy, lol. I will check if I can find a nice way to do this

Jonathan-Fang commented:
This is a great report to add! Looking forward to the end product!

joel-jeremy commented on packages/desktop-client/src/components/reports/graphs/CalendarGraph.tsx:
```diff
@@ -0,0 +1,314 @@
+      />
+      <span
+        style={{
+          fontSize: `${currentFontSize}px`,
```
Would `AutoTextSize` component work here?

lelemm replied:
![image](https://github.com/user-attachments/assets/00356b4a-a84e-4feb-b911-bea50920376f)
it would need a rework on the rest of the component to make it fit. its very big
but, after adjusting it each number will get its own size making it inconsistent sizing

joel-jeremy commented on packages/desktop-client/src/components/reports/graphs/CalendarGraph.tsx:
```diff
@@ -0,0 +1,314 @@
+        />
+      )}
+      <View
+        className="bar positive-bar"
```
We should just use `className` + emotion's `css` here to merge classnames and styles in one declaration

lelemm replied:
this is a left out from testing classes, removed the classnames

joel-jeremy commented on packages/desktop-client/src/components/reports/reports/Calendar.tsx:
```diff
@@ -0,0 +1,958 @@
+    onConditionsOpChange,
+  } = useFilters(widget?.meta?.conditions, widget?.meta?.conditionsOp);
+
+  useLayoutEffect(() => {
```
Can we use `useEffect` here?

joel-jeremy commented on packages/desktop-client/src/components/reports/reports/Calendar.tsx:
```diff
@@ -0,0 +1,958 @@
+    [navigate],
+  );
+
+  const CHEVRON_HEIGHT = 26;
```
We should declare these outside the component

lelemm replied:
done

joel-jeremy commented on packages/desktop-client/src/components/reports/reports/Calendar.tsx:
```diff
@@ -0,0 +1,958 @@
+    };
+  }, [data]);
+
+  const onSort = (headerClicked: string, ascDesc: 'asc' | 'desc') => {
```
Should event handler methods in this component be wrapped in useCallback?

joel-jeremy commented on packages/desktop-client/src/components/reports/reports/Calendar.tsx:
```diff
@@ -0,0 +1,958 @@
+
+  useEffect(() => {
+    if (refContainer.current) {
+      setTotalHeight(refContainer.current.clientHeight - 115);
```
We should put the 115 in a variable with meaningful name. Is it an offset?

lelemm replied:
done, its the size of the summary, so the expandable transactions don't hide it

joel-jeremy commented on packages/desktop-client/src/components/reports/reports/Calendar.tsx:
```diff
@@ -0,0 +1,958 @@
+    });
+  }, [totalHeight, mobileTransactionsOpen, api]);
+
+  const open = ({ canceled }: { canceled: boolean }) => {
```
Same comment above, should these be wrapped in useCallback?

lelemm replied:
done

joel-jeremy commented on packages/desktop-client/src/components/reports/reports/Calendar.tsx:
```diff
@@ -0,0 +1,958 @@
+          name="transactions"
+          items={[]}
+          fetchAllIds={async () => {
+            return Promise.resolve([]);
```
Do we actually need Promise.resolve here or can we just return an empty array?

lelemm replied:
changed

joel-jeremy commented on packages/desktop-client/src/components/reports/reports/Calendar.tsx:
```diff
@@ -0,0 +1,958 @@
+    </Page>
+  );
+}
+type CalendarHeaderProps = {
```
```suggestion
type CalendarWithHeaderProps = {
```

joel-jeremy commented on packages/desktop-client/src/components/reports/reports/Calendar.tsx:
```diff
@@ -0,0 +1,958 @@
+          )}
+        </View>
+      </View>
+      <div style={{ flexGrow: 1, marginBottom: 20 }}>
```
Does View not work here?

lelemm replied:
It works with display 'block', changed to it

joel-jeremy commented on packages/desktop-client/src/components/reports/reports/CalendarCard.tsx:
```diff
@@ -0,0 +1,551 @@
+} from '../spreadsheets/calendar-spreadsheet';
+import { useReport } from '../useReport';
+
+type CalendarProps = {
```
```suggestion
type CalendarCardProps = {
```

joel-jeremy commented on packages/desktop-client/src/components/reports/reports/CalendarCard.tsx:
```diff
@@ -0,0 +1,551 @@
+    return measurements;
+  }, []);
+
+  const debouncedResizeCallback = useRef(
```
Would useMemo work here?

lelemm replied:
changed to useMemo, it works properly

lelemm commented:
> > > It may be nice from a UX view to unselect the selected day in the full report if you click outside the calendar. Right now you have to remove the filter that gets generated.
> > 
> > 
> > yeah, nice one. I will do it
> 
> turns out, not easy, lol. I will check if I can find a nice way to do this

done @youngcw 


> This is looking very good! Thank you for your time and effort on this!
ty  @joel-jeremy  :)
> 
> I do have some comments on mobile view:
> 
> 1. When opening the filter in mobile, the whole page seems to re-render and the filter tooltip automatically closes
I noticed that too, I have to investigate what is going on. Any suspicion?

> 2. When the transaction list/drawer is open and you swipe down on the top of it, the drawer moves down to where your finger stops. I think swiping down should fully close the drawer.

changed how it works, take a look

> 3. Can we make the `Show transactions`/`Hide transactions` button larger so it's easier to click? Please also make it sentence case since we use sentence case throughout the app.
change the case and size of the button

> 4. Would it be easy to make the calendars line up vertically on mobile instead of horizontally so that we can utilize more screen real estate? (flexDirection = column for mobile?)
atm the calendar direction depends on how it shows on the desktop version, if height > width -> flex direction = column, otherwise, flex direction = row:
https://github.com/user-attachments/assets/e5034700-939a-4871-8575-c137d2b161f3





> Should we still display the Income and Expense summary as 0 when there are none for the selected days? ![image](https://private-user-images.githubusercontent.com/20313680/388203436-01348698-072a-4290-a560-83218c8cd72e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIxOTg1OTYsIm5iZiI6MTczMjE5ODI5NiwicGF0aCI6Ii8yMDMxMzY4MC8zODgyMDM0MzYtMDEzNDg2OTgtMDcyYS00MjkwLWE1NjAtODMyMThjOGNkNzJlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTIxVDE0MTEzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRjMmE1Y2E5ZTNmYmUwMjAzMDQyZTQ1MzA1MDhiMWJhNmY2NThkZDBhMzFjOTE0OTUzYmRkZjE3ZTRkYjc1NWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.nPtj7HwkhdZgxKFZcttjQ7b1nHEQFVaGgZVQVgGh9do)

removed the conditions to hide when zero

youngcw commented:
My only nitpik now is that the whole page shifts down when you add in a filter for the first time, which is a bit jarring.  Thats not that big of a deal for now though.

lelemm commented:
> My only nitpik now is that the whole page shifts down when you add in a filter for the first time, which is a bit jarring. Thats not that big of a deal for now though.

added a minHeight to the header, so as long as there is not too much filters, it will keep the screen static

joel-jeremy commented on packages/desktop-client/src/components/reports/reports/Calendar.tsx:
```diff
@@ -0,0 +1,953 @@
+                    alignItems: 'center',
+                  }}
+                >
+                  <Button
```
LGTM! One last nitpick, Can we remove the press highlight on this button? The text disappears when the button is clicked :)

lelemm replied:
do you mean like this?
![image](https://github.com/user-attachments/assets/3fc7cd1f-c2e1-48c5-9067-e40b1c5d5ae2)

joel-jeremy replied:
Yup, I think that would work :)

lelemm replied:
I committed it already.

what's up with this action?
Compare Sizes / compare (pull_request_target) Failing after 21s
mind explaining me when it fails?

matt-fidd replied:
It looked to be a flake, I've rerun and it passed now

joel-jeremy reviewed (approved):
LGTM! Would prefer another maintainer to approve before merging.

I think the next phase for the calendar report is to include preview transactions i.e. https://github.com/actualbudget/actual/issues/1928. Not sure if we want to do it here or in a separate PR. I'm fine either ways. :)

youngcw commented:
This needs the same update as the summary card to allow imnporting a dashboard.json file

lelemm commented:
> This needs the same update as the summary card to allow importing a dashboard.json file

I'm not sure what we need to do here. can you clarify please?

youngcw commented:
> > This needs the same update as the summary card to allow importing a dashboard.json file
> 
> I'm not sure what we need to do here. can you clarify please?

You need to copy what Mike did on the summary report

lelemm commented:
> > > This needs the same update as the summary card to allow importing a dashboard.json file
> > 
> > 
> > I'm not sure what we need to do here. can you clarify please?
> 
> You need to copy what Mike did on the summary report

ohh, got it!

done ✔️

lelemm commented:
@youngcw done. I guess I missed one filter =p

matt-fidd commented on packages/desktop-client/src/components/reports/reports/Calendar.tsx:
```diff
@@ -0,0 +1,959 @@
+                          fontStyle: 'italic',
+                        }}
+                      >
+                        No transactions
```
```suggestion
                        <Trans>No transactions</Trans>
```

matt-fidd commented on packages/desktop-client/src/components/reports/reports/Calendar.tsx:
```diff
@@ -0,0 +1,959 @@
+                    {!mobileTransactionsOpen && (
+                      <>
+                        <SvgCheveronUp width={16} height={16} />
+                        Show transactions
```
```suggestion
                        <Trans>Show transactions</Trans>
```

matt-fidd commented on packages/desktop-client/src/components/reports/reports/Calendar.tsx:
```diff
@@ -0,0 +1,959 @@
+                    {mobileTransactionsOpen && (
+                      <>
+                        <SvgCheveronDown width={16} height={16} />
+                        Hide transactions
```
```suggestion
                        <Trans>Hide transactions</Trans>
```

matt-fidd commented on packages/desktop-client/src/components/reports/reports/CalendarCard.tsx:
```diff
@@ -0,0 +1,543 @@
+                              marginRight: 4,
+                            }}
+                          >
+                            Income:
```
```suggestion
                            <Trans>Income:</Trans>
```

lelemm replied:
@jfdoming how would we do the privacy filter on this example?
`<Trans>Income: {{ incomeValue }}</Trans>`

jfdoming replied:
I think `<Trans>Income: <Privacy filter>{{ incomeValue }}</PrivacyFilter></Trans>` should work

jfdoming replied:
And generate a string like "Income: <1>{incomeValue}</1>"

lelemm replied:
@jfdoming  I kept @matt-fidd suggestion since Income and the value are not even on the same level:
![image](https://github.com/user-attachments/assets/a5f406d9-22bb-4094-b84b-076fd6457f62)

It may open for confusing context translations, but It keeps the layout as it is

jfdoming replied:
I see. Maybe in the future we may need to update, but for now I guess this is fine

matt-fidd commented on packages/desktop-client/src/components/reports/reports/CalendarCard.tsx:
```diff
@@ -0,0 +1,543 @@
+  type SetStateAction,
+  useCallback,
+} from 'react';
+import { useTranslation } from 'react-i18next';
```
```suggestion
import { Trans, useTranslation } from 'react-i18next';
```

matt-fidd commented on packages/desktop-client/src/components/reports/reports/CalendarCard.tsx:
```diff
@@ -0,0 +1,543 @@
+                              marginRight: 4,
+                            }}
+                          >
+                            Expenses:
```
```suggestion
                            <Trans>Expenses:</Trans>
```

matt-fidd reviewed (changes requested):
Just a couple of missed translations, everything else looks good

matt-fidd commented:
I just noticed the table is missing the parent transaction

Calendar:
![image](https://github.com/user-attachments/assets/539080b5-f325-43a4-b05e-6a538b78c97a)


Ledger:
![image](https://github.com/user-attachments/assets/25774f90-698e-465e-8ca5-d0bba53a63fc)

lelemm commented:
> I just noticed the table is missing the parent transaction
> 
> Calendar: ![image](https://private-user-images.githubusercontent.com/81489167/394980409-539080b5-f325-43a4-b05e-6a538b78c97a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM5Njk5NjIsIm5iZiI6MTczMzk2OTY2MiwicGF0aCI6Ii84MTQ4OTE2Ny8zOTQ5ODA0MDktNTM5MDgwYjUtZjMyNS00M2E0LWIwNWUtNmE1MzhiNzhjOTdhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjEyVDAyMTQyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNkYWM1ZDQ1NzA1NTk4Y2Y3M2EzOWExNDBmN2QwOWEyN2VkMDc0MmE0ODQyZGQ5ODIxMjVkODcxZjZkZTViOWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.USR-IDG_Y6X8x8ut8AGRR2hN3bsZeUGtH8XB_gXjlZY)
> 
> Ledger: ![image](https://private-user-images.githubusercontent.com/81489167/394980323-25774f90-698e-465e-8ca5-d0bba53a63fc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM5Njk5NjIsIm5iZiI6MTczMzk2OTY2MiwicGF0aCI6Ii84MTQ4OTE2Ny8zOTQ5ODAzMjMtMjU3NzRmOTAtNjk4ZS00NjVlLThjYTUtZDBiYmE1M2E2M2ZjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjEyVDAyMTQyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZmYzVjNmQ0N2YzZDVlOTZkOWJjZDA2OTEzZDU4MDMzNjM0MDBmYTUzZGY2MTExZDliNmQ0MWZiM2QxZWYyNDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.6ViEgxCVrKJGRk6vkzaokLs5yrZr63sQpjhnYliEH58)

any ideas how/why?

matt-fidd commented:
> > I just noticed the table is missing the parent transaction
> > Calendar: ![image](https://private-user-images.githubusercontent.com/81489167/394980409-539080b5-f325-43a4-b05e-6a538b78c97a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM5Njk5NjIsIm5iZiI6MTczMzk2OTY2MiwicGF0aCI6Ii84MTQ4OTE2Ny8zOTQ5ODA0MDktNTM5MDgwYjUtZjMyNS00M2E0LWIwNWUtNmE1MzhiNzhjOTdhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjEyVDAyMTQyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNkYWM1ZDQ1NzA1NTk4Y2Y3M2EzOWExNDBmN2QwOWEyN2VkMDc0MmE0ODQyZGQ5ODIxMjVkODcxZjZkZTViOWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.USR-IDG_Y6X8x8ut8AGRR2hN3bsZeUGtH8XB_gXjlZY)
> > Ledger: ![image](https://private-user-images.githubusercontent.com/81489167/394980323-25774f90-698e-465e-8ca5-d0bba53a63fc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM5Njk5NjIsIm5iZiI6MTczMzk2OTY2MiwicGF0aCI6Ii84MTQ4OTE2Ny8zOTQ5ODAzMjMtMjU3NzRmOTAtNjk4ZS00NjVlLThjYTUtZDBiYmE1M2E2M2ZjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjEyVDAyMTQyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZmYzVjNmQ0N2YzZDVlOTZkOWJjZDA2OTEzZDU4MDMzNjM0MDBmYTUzZGY2MTExZDliNmQ0MWZiM2QxZWYyNDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.6ViEgxCVrKJGRk6vkzaokLs5yrZr63sQpjhnYliEH58)
> 
> any ideas how/why?

Sorry, I'm not sure - it looks like the parent just isn't in the data

lelemm commented:
/update-vrt

lelemm commented:
> > > I just noticed the table is missing the parent transaction
> > > Calendar: ![image](https://private-user-images.githubusercontent.com/81489167/394980409-539080b5-f325-43a4-b05e-6a538b78c97a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM5Njk5NjIsIm5iZiI6MTczMzk2OTY2MiwicGF0aCI6Ii84MTQ4OTE2Ny8zOTQ5ODA0MDktNTM5MDgwYjUtZjMyNS00M2E0LWIwNWUtNmE1MzhiNzhjOTdhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjEyVDAyMTQyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNkYWM1ZDQ1NzA1NTk4Y2Y3M2EzOWExNDBmN2QwOWEyN2VkMDc0MmE0ODQyZGQ5ODIxMjVkODcxZjZkZTViOWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.USR-IDG_Y6X8x8ut8AGRR2hN3bsZeUGtH8XB_gXjlZY)
> > > Ledger: ![image](https://private-user-images.githubusercontent.com/81489167/394980323-25774f90-698e-465e-8ca5-d0bba53a63fc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM5Njk5NjIsIm5iZiI6MTczMzk2OTY2MiwicGF0aCI6Ii84MTQ4OTE2Ny8zOTQ5ODAzMjMtMjU3NzRmOTAtNjk4ZS00NjVlLThjYTUtZDBiYmE1M2E2M2ZjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjEyVDAyMTQyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZmYzVjNmQ0N2YzZDVlOTZkOWJjZDA2OTEzZDU4MDMzNjM0MDBmYTUzZGY2MTExZDliNmQ0MWZiM2QxZWYyNDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.6ViEgxCVrKJGRk6vkzaokLs5yrZr63sQpjhnYliEH58)
> > 
> > 
> > any ideas how/why?
> 
> Sorry, I'm not sure - it looks like the parent just isn't in the data

fixed

matt-fidd reviewed (approved):
LGTM, will leave to get another one/two reviews before merging

Great addition!
